src/solver/variable/economy/bindingConstraintsMarginalCost.h

```diff
@@ -323,12 +323,42 @@ class BindingConstMarginCost
         }
     }
 
+    void buildSurveyReport(SurveyResults& results,
+                           int dataLevel,
+                           int fileLevel,
+                           int precision) const
+    {
+        // Building syntheses results
+        // ------------------------------
+
+        // And only if we match the current data level _and_ precision level
+        if ((dataLevel & VCardType::categoryDataLevel) && (fileLevel & VCardType::categoryFileLevel)
+            && (precision & VCardType::precision))
+        {
+            results.isPrinted = isPrinted;
+            results.isCurrentVarNA[0] = isCurrentOutputNonApplicable(precision);
+            results.variableCaption = getBindConstraintCaption();
+
+            VariableAccessorType::template BuildSurveyReport<VCardType>(
+              results, pResults, dataLevel, fileLevel, precision);
+        }
+    }
+
 private:
+    // Private methods
+    // ---------------
     std::string getBindConstraintCaption() const
     {
         return associatedBC_->name + " (" + associatedBC_->operatorType + ")";
     }
 
+    bool isCurrentOutputNonApplicable(int precision) const
+    {
+        return precision < pow(2, associatedBC_->type - 1);
```

Comment:
```suggestion
        // Explanatory comment
        return precision < (1 << (associatedBC_->type - 1));
```

Reply:
Also check that `associatedBC_->type > 0`. 
```cpp
    enum Type
    {
        //! Unknown status
        typeUnknown = 0,
        //! The binding constraint has hourly values
        typeHourly,
        //! The binding constraint has daily values
        typeDaily,
        //! The binding constraint has weekly values
        typeWeekly,
        //! The maximum number of types
        typeMax
    };
```

Reply:
Already done in : **src\libs\antares\study\constraint\constraint.cpp** : 760